### DISCUSSION TITLE
Let `action` return 401 or 403

### DISCUSSION THREAD

**Discussion by tats-u** [edmundhung/conform#814] [Q&A] [answered]
This library stop working when `action` returns 401, 403, or other errors, meaning the login failure, the detection of account lock, the maintenance mode, etc.
Returning without such error codes is not best because it's not "OK".
Especially, `remix-auth-form` force `action` to returns 401 or redirects to a page (including the same page).

**Comment by edmundhung** [comment]
Conform does not care about the status code as long as it receives the submission result back from the server. You might need to share a repo so someone can take a better look.

**Reply by tats-u** [reply]
We might have to use ErrorBoundary. Thank you.

**Reply by tats-u** [reply, marked as answer]
Worked well by:

```js
return data(submission.reply({formErros: ["ID or password incorrect."]}), 401);
```

- Don't throw, just return.
- `submission.reply` returns just a JSON-compatible object.
- I tried it in an environment where the v3_singleFetch is enabled. This is why I used `data` instead of `json`. I have never tried in those without it.
